modules/run-vault/run-vault

```diff
@@ -226,32 +225,55 @@ EOF
   chown "$user:$user" "$config_path"
 }
 
-function generate_supervisor_config {
-  local -r supervisor_config_path="$1"
+function generate_systemd_config {
+  local -r systemd_config_path="$1"
   local -r vault_config_dir="$2"
   local -r vault_bin_dir="$3"
-  local -r vault_log_dir="$4"
-  local -r vault_log_level="$5"
-  local -r vault_user="$6"
-
-  log_info "Creating Supervisor config file to run Vault in $supervisor_config_path"
-  cat > "$supervisor_config_path" <<EOF
-[program:vault]
-command=$vault_bin_dir/vault server -config $vault_config_dir -log-level=$vault_log_level
-stdout_logfile=$vault_log_dir/vault-stdout.log
-stderr_logfile=$vault_log_dir/vault-error.log
-numprocs=1
-autostart=true
-autorestart=true
-stopsignal=INT
-user=$vault_user
+  local -r vault_log_level="$4"
+  local -r vault_user="$5"
+  local -r config_path="$config_dir/$VAULT_CONFIG_FILE"
+
+  log_info "Creating systemd config file to run Vault in $systemd_config_path"
+  cat > "$systemd_config_path" <<EOF
+[Unit]
+Description="HashiCorp Vault - A tool for managing secrets"
+Documentation=https://www.vaultproject.io/docs/
+Requires=network-online.target
+After=network-online.target
+ConditionFileNotEmpty=$config_path
+
+[Service]
```

Comment:
The `supervisord` config set the stdout and stderr log files explicitly. Where are the logs going with systemd? Are we assuming users will know to use `journalctl`? Should that be documented somewhere?

Reply:
Added documentation and the `systemd-stdout` and `systemd-stderr` options